app/src/main/java/com/example/tanya/godeltechchallengeandroid/ui/start/StartPresenter.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.tanya.godeltechchallengeandroid.ui.start
+
+import android.os.Handler
+import com.example.tanya.godeltechchallengeandroid.util.Prefs
+import javax.inject.Inject
+
+class StartPresenter
+    @Inject
+    constructor(private val prefs: Prefs): StartContract.Presenter {
+
+    private lateinit var view: StartContract.View
+
+    override fun bindView(view: StartContract.View) {
+        this.view = view
+    }
+
+    /*Imitating data loading*/
+    override fun loadData() {
+        Handler().postDelayed(
```

Comment:
Why didn't you use RxJava Observable.delay()?

According to the clean architecture approach the presenter and other classes (except on views, views controllers, and android class wrapper) know nothing about android classes. Only java and kotlin.

Reply:
Also I can see you didn't handle the screen rotation. How do you think you will receive an exception if you will rotate your device or not?)